metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -299,7 +299,11 @@ final class BloopServers(
                 requested != maybeBloopGlobalJsonJvmProperties
               )
           ) Some(BloopJsonUpdateCause.JVM_OPTS)
-          else if (maybeRequestedMetalsJavaHome != maybeBloopGlobalJsonJavaHome)
+          else if (
+            maybeRequestedMetalsJavaHome.exists { metalsJavaHome =>
+              maybeBloopGlobalJsonJavaHome.exists(_ != metalsJavaHome)
+            }
+          )
```

Comment:
We might need to update `.bloop/bloop.json` if `.bloop/bloop.json` doesn't have `javaHome` and Metals' `bloop java home` has a value 🤔 

Summary:

|     |      maybeGlobalJsonJavaHome = None      |  maybeGlobalJsonJavaHome=Some(value) |
|----------|:-------------:|------:|
| maybeRequestedMetalsJavaHome=None |  nothing happens | nothing happens |
| maybeRequestedMetalsJavaHome=Some(value) |    should write value to json |  compare version and if they're different, write value to json  |

Reply:
Ah, good point. I'll update this.

Reply:
So, I misunderstood `maybeRequestedMetalsJavaHome` is always `Some(value)` in https://github.com/scalameta/metals/pull/3969#discussion_r887650732 but it is the case only with `metals-vscode`...

Reply:
FYI @tgodzik

Reply:
Ahhh, I totally missed that it's always set with VS Code. Yea, that won't happen in other clients.